File: php/number1.php
<?php

$m = 3;
$n = 3;
$s = "ULDDRRUU";

function startPosFinder($m, $n, $s) {
    $x = 0;
    $y = 0;
    $rX = 0;
    $lX = 0;
    $uY = 0;
    $dY = 0;

    for ($i = 0; $i < strlen($s); $i++) {
        if ($s[$i] == "D") {
            $y += 1;
            if ($y > $dY) {
                $dY = $y;
            }
        } elseif ($s[$i] == "U") {
            $y -= 1;
            if ($y < $uY) {
                $uY = $y;
            }
        } elseif ($s[$i] == "L") {
            $x -= 1;
            if ($x < $lX) {
                $lX = $x;
            }
        } elseif ($s[$i] == "R") {
            $x += 1;
            if ($x > $rX) {
                $rX = $x;
            }
        }
    }

    if ($rX - $lX >= $m || $dY - $uY >= $n) {
        return "No";
    }
    return [1 - $lX, 1 - $uY];
}

print_r(startPosFinder($m, $n, $s));

?>

File: cs/number1.cs
using System;

class Program
{
    static void StartPosFinder(int m, int n, string s)
    {
        int x = 0;
        int y = 0;
        int rx = 0;
        int lx = 0;
        int uy = 0;
        int dy = 0;

        foreach (char c in s)
        {
            if (c == 'U')
            {
                y--;
                if (y < uy) uy = y;
            }
            else if (c == 'D')
            {
                y++;
                if (y > dy) dy = y;
            }
            else if (c == 'L')
            {
                x--;
                if (x < lx) lx = x;
            }
            else if (c == 'R')
            {
                x++;
                if (x > rx) rx = x;
            }
        }

        if (rx - lx >= m || dy - uy >= n)
        {
            Console.WriteLine("No");
        }
        else
        {
            Console.WriteLine($"{1 - lx} {1 - uy}");
        }
    }

    static void Main()
    {
        string s = "ULDDRRUU";
        StartPosFinder(3, 3, s);
    }
}
